lib/generateClass.js

```diff
@@ -284,14 +307,222 @@ function writeToJson(def, cw) {
+
+                if (element[fieldName]) {
+                  fixedValue = element[fieldName];
+                }
+              }
+
+              if (!fixedValue) {
+                console.error('Could not identify fixed value for:');
+                console.error(element);
+              }
+
+              sliceGroup.elements[sliceGroup.currentSlice].unshift( { path: element.path, value: fixedValue } );
+            }
+          });
+        } 
+      }
+    }
+
+    // if the fhir profile ends and we still have "open" slices, finalize them here
+    while (sliceStack.length > 0) {
+      const sliceGroup = sliceStack.shift();
+      sliceMap[sliceGroup.path] = sliceGroup.elements;
+    }
+
+    for (let element of fhirProfile.snapshot.element) {
+      const mapping = element.mapping && element.mapping.find(m => m['identity'] === 'shr');
+      let baseIsList = element.base && element.base.max == '*';
+      baseIsList = baseIsList || (element.mapping && element.mapping.filter(elem => elem.identity == 'shr').length > 1);
+      if(mapping !== undefined){
+        if(mapping.map === '<Value>'){
+          // Mapping to the value of this es6 instance
+          if (def.value instanceof IdentifiableValue && def.value.identifier.isPrimitive) {
+            generateFromFHIRAssignment(def.value.card.isList, baseIsList, 1, element, [], 'value', fhirProfile, null, cw);
+          } else {
+            console.error('ERROR: Value referenced in mapping but none exist on this element.');
+          }
+        } else {
+          // Mapping to a field within this es6 instance
+          const fieldMapPath = mapping.map.match(/<([^>]*)>/g);
+          if(fieldMapPath){
+            // Generate a chain if mapped multiple levels deep, such as myField1.subField2
+            const classMethodChain = fieldMapPath.map(e => toSymbol(fhirMappingToIdentifier(e).name)).join('.');
+            // find related field if available
+            let field;
+            let currDef = def;
+
+            // for nested element mappings, such as "Patient.person.address", we have to look up Patient.person and then Person.address
+            for(const pathElement of fieldMapPath) {
+              field = currDef.fields.find(f => f.identifier && f.identifier.equals(fhirMappingToIdentifier(pathElement)));
+
+              if (!field) {
+                break;
+              }
+
+              currDef = specs.dataElements.byNamespace(field.identifier.namespace).find(e =>  e.identifier.equals(field.identifier));
+            }
+
+            if(field){
+              // console.log(field.identifier.name);
+              if (!alreadyMappedElements.has(classMethodChain)) {
+                const constraintsLength = field.constraintsFilter.includesType.constraints.length;
+                const slicing = sliceMap[element.path] && sliceMap[element.path][element.sliceName];
```

Comment:
Are you sure the `&&` converts it to true/false? I'm intending to use this as a "guard operator" where it will only try to access `sliceMap[element.path][element.sliceName]` if `sliceMap[element.path]` is not null. I figured the one line version is shorter than the traditional null check.

Reply:
But I definitely have to look into whether `element.sliceName` will always be present here. From the examples I saw the SHR mapping was always on the same element as the sliceName, but maybe that won't necessarily be the case.

Reply:
Wow.  You are 100% right, @dehall .  TIL that I've been misunderstanding `&&` in JavaScript for a _long_ time.  Thanks for setting me straight!